internal/credential/vault/vault.go

```diff
@@ -0,0 +1,108 @@
+package vault
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/go-rootcerts"
+	vault "github.com/hashicorp/vault/api"
+)
+
+type clientConfig struct {
+	Addr                  string
+	Token                 string
+	CaCert                []byte
+	ClientCert, ClientKey []byte
+	TlsServerName         string
+	TlsSkipVerify         bool
+	Namespace             string
+}
+
+func (c *clientConfig) IsValid() bool {
+	if c == nil || c.Addr == "" || c.Token == "" {
+		return false
+	}
+	return true
+}
+
+func (c *clientConfig) IsClientTLS() bool {
+	if len(c.ClientCert) > 0 && len(c.ClientKey) > 0 {
+		return true
+	}
+	return false
+}
+
+type client struct {
+	cl    *vault.Client
+	token string
+}
+
+func newClient(c *clientConfig) (*client, error) {
+	const op = "vault.newClient"
+	if !c.IsValid() {
+		return nil, errors.New(errors.InvalidParameter, op, "invalid configuration")
+	}
+	vc := vault.DefaultConfig()
+	vc.Address = c.Addr
+	if len(c.CaCert) > 0 {
+		rootConfig := &rootcerts.Config{
+			CACertificate: []byte(c.CaCert),
+		}
+		tlsConfig := vc.HttpClient.Transport.(*http.Transport).TLSClientConfig
```

Comment:
Since you have the `TlsSkipVerify` bool in the config would you not want to set`InsecureSkipVerify` here?

Reply:
Great catch! Ty!